### PR TITLE
Use java 8 features in TierSelectorStrategy implementations

### PR DESCRIPTION
**Pull request by drcrallen**

### CODE REVIEW

File: server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java [hunk 1]
```diff
@@ -47,6 +39,6 @@ public HighestPriorityTierSelectorStrategy(@JacksonInject ServerSelectorStrategy
   @Override
   public Comparator<Integer> getComparator()
   {
-    return comparator;
+    return REVESE_COMPARATOR;
```
Comment (leventov) [inline server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java]: It's `Comparator.reverseOrder()`

File: server/src/main/java/io/druid/client/selector/LowestPriorityTierSelectorStrategy.java [hunk 2]
```diff
@@ -47,6 +37,6 @@ public LowestPriorityTierSelectorStrategy(@JacksonInject ServerSelectorStrategy
   @Override
   public Comparator<Integer> getComparator()
   {
-    return comparator;
+    return Integer::compare;
```
Comment (leventov) [inline server/src/main/java/io/druid/client/selector/LowestPriorityTierSelectorStrategy.java]: It's `Comparator.naturalOrder()`

File: server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java [hunk 3]
```diff
@@ -21,22 +21,14 @@
...
 {
-  private static final Comparator<Integer> comparator = new Comparator<Integer>()
-  {
-    @Override
-    public int compare(Integer o1, Integer o2)
-    {
-      return Ints.compare(o2, o1);
-    }
-  };
+  private static final Comparator<Integer> REVESE_COMPARATOR = (o1, o2) -> Integer.compare(o2, o1);
```
Comment (leventov) [inline server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java]: This constant is not needed

File: server/src/main/java/io/druid/client/selector/CustomTierSelectorStrategy.java [hunk 4]
```diff
@@ -48,15 +48,10 @@ public CustomTierSelectorStrategy(
...
-    this.comparator = new Comparator<Integer>()
-    {
-      @Override
-      public int compare(Integer o1, Integer o2)
-      {
-        int pos1 = lookup.get(o1);
-        int pos2 = lookup.get(o2);
-        return Ints.compare(pos1, pos2);
-      }
+    this.comparator = (o1, o2) -> {
```
Comment (leventov) [inline server/src/main/java/io/druid/client/selector/CustomTierSelectorStrategy.java]: It's `Comparator.comparingInt(o -> lookup.get(o))`

File: server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java (hunk 1 above)
Comment (drcrallen) [reply server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java]: I couldn't get `Integer::compare.reverseOrder()` or similar to work. Every kind of syntax fixing ended up being much dirtier looking compared to the way it is here. I'm open to suggestions though.

Comment (drcrallen) [reply server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java]: probably natural order, let me try that one

Comment (leventov) [reply server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java]: It's a polymorphic static method, https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html#reverseOrder--, you should be able to write just `Comparator.reverseOrder()`.

File: server/src/main/java/io/druid/client/selector/CustomTierSelectorStrategy.java (hunk 4 above)
Comment (drcrallen) [reply server/src/main/java/io/druid/client/selector/CustomTierSelectorStrategy.java]: `Comparator.comparingInt(lookup::get)` fixed

File: server/src/main/java/io/druid/client/selector/LowestPriorityTierSelectorStrategy.java (hunk 2 above)
Comment (drcrallen) [reply server/src/main/java/io/druid/client/selector/LowestPriorityTierSelectorStrategy.java]: fixed

File: server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java (hunk 1 above)
Comment (drcrallen) [reply server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java]: fixed

File: server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java (hunk 3 above)
Comment (drcrallen) [reply server/src/main/java/io/druid/client/selector/HighestPriorityTierSelectorStrategy.java]: fixed

**Pull request merged**